airflow/datasets/metadata.py

```diff
@@ -0,0 +1,39 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import attrs
+
+from airflow.datasets import coerce_to_uri
+
+if TYPE_CHECKING:
+    from airflow.datasets import Dataset
+
+
+@attrs.define(init=False)
+class Metadata:
```

Comment:
Where in this PR is the new `Metadata` class actually being used? Was looking for it but could not see a reference. Noth that it is "fresh dead code"?

Reply:
Okay, took a while to find it... nevermind.